e2e/testcases/override_reconcile_timeout_test.go

```diff
@@ -40,6 +41,19 @@ func TestOverrideReconcileTimeout(t *testing.T) {
 	// Only actuation should succeed, reconcile should time out.
 	nt.MustMergePatch(rootSync, `{"spec": {"override": {"reconcileTimeout": "30s"}}}`)
 	nt.WaitForRepoSyncs()
+	// Scale up the Autopilot cluster to prevent the pod used for test from failing to schedule
+	// TODO: remove this after resource group controller updated to update the reconcile status
+	if nt.IsGKEAutopilot {
+		nt.MustKubectl("apply", "-f", "../testdata/deployment-helloworld.yaml")
+		_, err := nomostest.Retry(nt.DefaultWaitTimeout, func() error {
```

Comment:
You can use the new WatchForObject with the StatusEquals Current Precondition.

Reply:
Sounds good, changed to use the new WatchForObject.